src/nwb_conversion_tools/tools/hdmf.py

```diff
@@ -90,3 +92,85 @@ def _get_maxshape(self) -> tuple:
 
     def _get_data(self, selection: Tuple[slice]) -> np.ndarray:
         return self.data[selection]
+
+
+class ImagingExtractorDataChunkIterator(GenericDataChunkIterator):
+    """
+    Generic data chunk iterator for an ImagingExtractor object
+    primarily used when writing imaging data to an NWB file.
+    """
+
+    def __init__(
+        self,
+        imaging_extractor: ImagingExtractor,
+        buffer_gb: Optional[float] = None,
+        buffer_shape: Optional[tuple] = None,
+        chunk_mb: Optional[float] = None,
+        chunk_shape: Optional[tuple] = None,
+        display_progress: bool = False,
+        progress_bar_options: Optional[dict] = None,
+    ):
+        self.imaging_extractor = imaging_extractor
+
+        if chunk_mb is None and chunk_shape is None:
+            chunk_mb = 1.0
+
+        self._maxshape = self._get_maxshape()
+        self._dtype = self._get_dtype()
+        if chunk_shape is None:
+            chunk_shape = super()._get_default_chunk_shape(chunk_mb=chunk_mb)
+
+        if buffer_gb is None and buffer_shape is None:
+            buffer_gb = 1.0
+
+        if buffer_shape is None:
+            buffer_shape = self._get_scaled_buffer_shape(buffer_gb=buffer_gb, chunk_shape=chunk_shape)
+        else:
+            assert buffer_shape[1:] == self._maxshape[1:], (
+                f"Except from the first axis, the buffer shape ({self.buffer_shape}) "
+                f"must be equal to max shape ({self._maxshape})."
+            )
+
+        if buffer_shape[0] % chunk_shape[0] != 0:
+            # warn if the first axis is not a multiple of the chunk shape
+            warn(
+                f"The first axis of the buffer shape ({buffer_shape[0]}) must be divisible by the first axis of the chunk shape ({chunk_shape[0]})"
+            )
+            # override the chunk shape?
+            chunk_shape = tuple([chunk_shape[0] - (buffer_shape[0] % chunk_shape[0])]) + chunk_shape[1:]
```

Comment:
I wouldn't override it magically in the event the shape isn't correct - it should be set correctly either manually, or with 100% reliability from the default